pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -143,6 +143,21 @@ func New(args runtime.Object, handle frameworktypes.Handle) (frameworktypes.Plug
 		})
 	}
 
+	if defaultEvictorArgs.MinReplicas > 1 {
+		podLister := handle.SharedInformerFactory().Core().V1().Pods().Lister()
```

Comment:
Any chance to list pods by owner UID? E.g. creating a new cache that will get updated based on pod CRUD ops and allow to quickly get a list of pods owned by an owner UID? In huge clusters this might be a bottle neck.

Reply:
that's a really good suggestion, let me take a stab at it and see how that works

Reply:
I tried this and the indexer only supports a single key to index by, while ownerReferences is an array.

Reply:
https://github.com/kubernetes-sigs/descheduler/blob/master/pkg/descheduler/pod/pods.go#L131 can return a list of keys. In case the implementation ends up comparing slice with a slice you can always get a list of all pods owned by first owner, second, third and make an intersection of all the lists.

Reply:
What do you think is the performance gain here? Loading too much cached objects in memory?

Both approach are already using the informer cache so it's no extra lookup to the apiserver.

Reply:
Iterating through too many pods in huge clusters. Though, we can merge the code as it is and improve it once there's a request for it I guess.

Reply:
It's already namespaced. I could limit it further by adding the following labels:
- `pod-template-hash`, if it exists
- `controller-revision-hash`, if it exists

Thoughts? I'll take another look at indexer again

Reply:
Right. I missed that namespaced part :) Fine then :)